operations/iuf/workflows/upgrade_all_products.md

```diff
@@ -1,120 +1,577 @@
+
+    (`ncn-m001#`) Execute the `management-nodes-rollout` stage on all remaining worker and master nodes.
+
+    ```bash
+    iuf -a "${ACTIVITY_NAME}" run -r management-nodes-rollout
+    ```
+
+1. Use `kubectl` to remove the `iuf-prevent-rollout=true` label from the canary node. Replace the example value of `${XNAME}` with the xname of the canary node.
+
+    ```bash
+    XNAME=x3000c0s29b1n0
+    kubectl label node "${XNAME}" --overwrite iuf-prevent-rollout-
+    ```
+
+Once this step has completed:
+
+- Management NCN worker nodes have been rebuilt with the image and CFS configuration created in previous steps of this workflow
+- Per-stage product hooks have executed for the `management-nodes-rollout` stage
+
+#### NCN master nodes
+
+Unlike NCN worker nodes, NCN master nodes do not contain kernel module content from non-CSM products. However, userspace non-CSM product content is still provided on NCN master nodes and thus the `prepare-images` stage creates a
+new image and CFS configuration for NCN master nodes. The CFS configuration layers ensure the non-CSM product content is applied correctly for both image customization and node personalization scenarios. As a result, the administrator
+can decide how to apply the new content to the NCN master nodes by following **one** of the following procedures:
+
+1. Rebuild the NCN master nodes (rebooting to the new image and CFS configuration)
+
+   If this procedure is used, follow the procedure documented in [NCN worker nodes](#ncn-worker-nodes) to rebuild the master nodes. For all invocations of `iuf run -r management-nodes-rollout`, also specify
+   `--limit-management-rollout Management_Master` to limit the rebuild operations to the NCN master nodes.
+
+   **`NOTE`** the `management-nodes-rollout` stage currently does not rebuild `ncn-m001`. Follow the instructions in [Stage 2.3 - `ncn-m001` upgrade](../../../upgrade/Stage_2.md#stage-23---ncn-m001-upgrade) to rebuild `ncn-m001` outside of IUF.
+
+1. Update the CFS configuration (applying the new CFS configuration to the existing node without rebuilding)
+
+   If this procedure is used, update the CFS configuration for all of the NCN master nodes by following the instructions documented in
+   [Update a CFS Configuration](../../configuration_management/Update_a_CFS_Configuration.md#update-a-cfs-configuration).
+
+Once this step has completed:
+
+- Management NCN master nodes have either been rebuilt with the image and CFS configuration created in previous steps of this workflow or have had their CFS configuration updated to the CFS configuration created in previous steps
+  of this workflow
+- Per-stage product hooks have executed for the `management-nodes-rollout` stage if the master nodes were rebuilt
+
+#### NCN storage nodes
+
+IUF currently does not manage the rebuild or NCN storage nodes. Follow the instructions in [Stage 1 - Ceph image upgrade](../../../upgrade/Stage_1.md) to rebuild the NCN storage nodes.
```

Comment:
We might mention we build and customize the storage node.  "While IUF builds and customizes an image for the storage nodes when used with the provided bootprep config files in the HPC CSM Software Recipe, ..."

Reply:
@leliasen-hpe added a commit to expand on and correct the storage node text.  There's still 2 ```TODO``` placeholders there for details she is working out, be see if the updated text looks better and let us know.